pulp_ansible/app/galaxy/v3/filters.py

```diff
@@ -0,0 +1,112 @@
+from django.contrib.postgres.search import SearchQuery
+from django.db.models import fields as db_fields
+from django.db.models.expressions import F, Func
+from django_filters import filters
+from django_filters import FilterSet
+from django.db.models import Q
+
+
+class CollectionVersionSearchFilter(FilterSet):
+    """A custom filterset for cross-repo search."""
+
+    name = filters.CharFilter(method="filter_by_name")
+    namespace = filters.CharFilter(method="filter_by_namespace")
+    version = filters.CharFilter(method="filter_by_version")
+    distribution = filters.CharFilter(method="filter_by_repository_name")
+    repository = filters.CharFilter(method="filter_by_repository_name")
+    repository_name = filters.CharFilter(method="filter_by_repository_name")
+    dependency = filters.CharFilter(method="filter_by_dependency")
+    deprecated = filters.CharFilter(method="filter_by_deprecated")
+    signed = filters.CharFilter(method="filter_by_signed")
+    is_highest = filters.BooleanFilter(field_name="is_highest")
+    q = filters.CharFilter(field_name="q", method="filter_by_q")
+    keywords = filters.CharFilter(field_name="q", method="filter_by_q")
+
+    tags = filters.CharFilter(
+        field_name="tags",
+        method="filter_by_tags",
+        # help_text=_("Filter by comma separate list of tags that must all be matched"),
+        help_text="Filter by comma separate list of tags that must all be matched",
+    )
+
+    def filter_by_repository_name(self, qs, name, value):
+        """Allow for multiple repository names to filter on."""
+
+        # print(f'QS:{qs} NAME:{name} VALUE:{value}')
+
+        include_q = Q()
+        if "," in value:
+            repository_names = value.split(",")
+            for rn in repository_names:
+                include_q = include_q | Q(repository__name=rn)
+        else:
+            include_q = include_q | Q(repository__name=value)
+        qs = qs.filter(include_q)
+        return qs
+
+    def filter_by_name(self, qs, name, value):
+        return qs.filter(Q(collectionversion__name=value))
+
+    def filter_by_namespace(self, qs, name, value):
+        return qs.filter(Q(collectionversion__namespace=value))
+
+    def filter_by_version(self, qs, name, value):
+        return qs.filter(Q(collectionversion__version=value))
+
+    def filter_by_dependency(self, qs, name, value):
+        """Return a list of collections that depend on a given collection name."""
+        qs = qs.filter(collectionversion__dependencies__has_key=value)
```

Comment:
These are also available on the model, I don't think we need to look through the collection version model.